src/main/java/eu/xenit/transformer/arender/alfresco/content/transformer/ARenderContentTransformerWorker.java

```diff
@@ -0,0 +1,55 @@
+package eu.xenit.transformer.arender.alfresco.content.transformer;
+
+import java.io.InputStream;
+import java.util.UUID;
+import org.alfresco.repo.content.transform.ContentTransformerWorker;
+import org.alfresco.service.cmr.repository.ContentReader;
+import org.alfresco.service.cmr.repository.ContentWriter;
+import org.alfresco.service.cmr.repository.TransformationOptions;
+import org.apache.log4j.Logger;
+
+public class ARenderContentTransformerWorker implements ContentTransformerWorker {
+
+    private static final Logger LOGGER = Logger.getLogger(ARenderContentTransformerWorker.class);
+
+    private ContentTransformARenderRestClient restClient;
+
+    @Override
+    public boolean isAvailable() {
+        return restClient.getWeatherPerformance() >= 0;
+    }
+
+    @Override
+    public String getVersionString() {
+        return "1.0.0";
+    }
+
+    @Override
+    public boolean isTransformable(String sourceMimetype, String targetMimetype, TransformationOptions options) {
+        return "application/pdf".equals(targetMimetype);
+    }
+
+    @Override
+    public String getComments(boolean available) {
+        return null;
+    }
+
+    @Override
+    public void transform(ContentReader reader, ContentWriter writer, TransformationOptions options) throws Exception {
+        String uuid = UUID.randomUUID().toString();
```

Comment:
Maybe it would also be nice to have some basic information of document linked to the uuid (filename, noderef, ...) in a first log line.

Reply:
Unfortunately, that is not accessible to the transformer. The best you get is the ContentReader & ContentWriter (when calling toString() on it, it gives you the contentUrl, size, mimetype)

Reply:
Also, the information is logged on line 41, reader together with the UUID.